dali/test/python/test_RN50_data_fw_iterators.py

```diff
@@ -0,0 +1,177 @@
+
+class RN50Pipeline(Pipeline):
+    def __init__(self, batch_size, num_threads, device_id, num_gpus, data_paths, prefetch, fp16, nhwc):
+        super(RN50Pipeline, self).__init__(batch_size, num_threads, device_id, prefetch_queue_depth=prefetch)
+        self.input = ops.FileReader(file_root = data_paths[0], shard_id = device_id, num_shards = num_gpus)
+        self.decode_gpu = ops.nvJPEGDecoder(device = "mixed", output_type = types.RGB)
+        self.res = ops.RandomResizedCrop(device="gpu", size =(224,224))
+
+        layout = types.NHWC if nhwc else types.NCHW
+        out_type = types.FLOAT16 if fp16 else types.FLOAT
+
+        self.cmnp = ops.CropMirrorNormalize(device="gpu",
+                                            output_dtype=out_type,
+                                            output_layout=layout,
+                                            crop=(224, 224),
+                                            image_type=types.RGB,
+                                            mean=[0.485 * 255,0.456 * 255,0.406 * 255],
+                                            std=[0.229 * 255,0.224 * 255,0.225 * 255])
+        self.coin = ops.CoinFlip(probability=0.5)
+
+    def define_graph(self):
+        rng = self.coin()
+        jpegs, labels = self.input(name="Reader")
+        images = self.decode_gpu(jpegs)
+        images = self.res(images)
+        output = self.cmnp(images.gpu(), mirror=rng)
+        return (output, labels.gpu())
+
+parser = argparse.ArgumentParser(description='Test RN50 augmentation pipeline with different FW iterators')
+parser.add_argument('-g', '--gpus', default=1, type=int, metavar='N',
+                    help='number of GPUs (default: 1)')
+parser.add_argument('-b', '--batch', default=2048, type=int, metavar='N',
+                    help='batch size (default: 2048)')
+parser.add_argument('-p', '--print-freq', default=10, type=int,
+                    metavar='N', help='print frequency (default: 10)')
+parser.add_argument('-j', '--workers', default=3, type=int, metavar='N',
+                    help='number of data loading workers (default: 3)')
+parser.add_argument('--prefetch', default=2, type=int, metavar='N',
+                    help='prefetch queue deptch (default: 2)')
+parser.add_argument('--fp16', action='store_true',
+                    help='Run fp16 pipeline')
+parser.add_argument('--nhwc', action='store_true',
+                    help='Use NHWC data instead of default NCHW')
+parser.add_argument('-i', '--iters', default=-1, type=int, metavar='N',
+                    help='Number of iterations to run (default: -1 - whole data set)')
+parser.add_argument('-e', '--epochs', default=1, type=int, metavar='N',
+                    help='Number of epochs to run (default: 1)')
+args = parser.parse_args()
+
+N = args.gpus             # number of GPUs
+BATCH_SIZE = args.batch   # batch size
+LOG_INTERVAL = args.print_freq
+WORKERS = args.workers
+PREFETCH = args.prefetch
+FP16 = args.fp16
+NHWC = args.nhwc
```

Comment:
Maybe above lines could be removed and `args` used everywhere. I don't think they introduce anything useful right now.

Reply:
will do